### PR TITLE
perf(utils): better implementation of repeat_expand_2d

### PR DESCRIPTION
**Pull request by BlueAmulet**
After doing some profiling I found that repeat_expand_2d is relatively slow. Rather than manually repeating the array, take advantage of torch.nn.functional.interpolate which is a couple orders of magnitude faster.

I have tested the new and old methods and found the results to be the same.

### CODE REVIEW

**Comment by 34j** [comment]
By the way, I'm really worried about `f0.interpolate_f0`. Like it obviously shouldn't be called at runtime.
